<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Intro to RBioTools</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RBiotools</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    About
  </a>
</li>
<li>
  <a href="getting-started.html">
    <span class="fa fa-info"></span>
     
    Getting Started
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Intro to RBioTools</h1>

</div>


<p>RBiotools is a package for COMPARATIVE MICROBIAL GENOMICS. RBiotools
works best in the RStudio integrated development environment. RBiotools
works with IDs from GenBank, the NIH genetic sequence data base, to
perform analyses and create figures.</p>
<div id="install-r-r-studio-and-other-setup" class="section level1"
number="1">
<h1><span class="header-section-number">1</span> Install R, R Studio,
and other setup</h1>
<p>To get started there are several installations that must be
completed. These items are listed in the order in which they should be
completed.</p>
<ol style="list-style-type: decimal">
<li><p>Install R from here: <a href="https://www.r-project.org/"
class="uri">https://www.r-project.org/</a></p>
<ul>
<li>Make sure that you install the correct version for your operating
system.</li>
</ul></li>
<li><p>Install R Studio from here: <a href="https://www.rstudio.com/"
class="uri">https://www.rstudio.com/</a></p>
<ul>
<li>Make sure that you install the correct version for your operating
system.</li>
</ul></li>
<li><p>Install requirements for compiling R packages</p>
<ul>
<li><p>If Windows, Install Rtools4 <a
href="https://cran.r-project.org/bin/windows/Rtools/rtools40.html"
class="uri">https://cran.r-project.org/bin/windows/Rtools/rtools40.html</a></p></li>
<li><p>If Mac, Install Xcode Command Line Tools</p>
<ul>
<li><p>i. Open a terminal window (Click Spotlight search in the top
right of your screen, then search for “Terminal”)</p></li>
<li><p>ii. Copy and paste the following into the terminal
<code>xcode-select --install</code>, then press Enter. You may need to
provide your password to enable installing the software.</p></li>
<li><p>iii. Follow any onscreen instructions and wait for it to
finish.</p></li>
</ul></li>
</ul></li>
<li><p>Designate a directory on your computer where you will store your
work, this will be your <strong>Working Directory</strong>.</p>
<ul>
<li><em>Suggestion:</em> Create a directory called BMIG5101 where you
can store all of your class materials for the semester.</li>
</ul></li>
<li><p>Download the RBioTools tar.gz from here: <a
href="https://github.com/nsf-dart/RBioTools/blob/91a28f0bc261617280a63ef0dbd4b712cbfb0200/docs/RBiotools_0.5.3.tar.gz"
class="uri">https://github.com/nsf-dart/RBioTools/blob/91a28f0bc261617280a63ef0dbd4b712cbfb0200/docs/RBiotools_0.5.3.tar.gz</a></p>
<ul>
<li>Save this tar.gz file in the root of your Working Directory</li>
</ul></li>
<li><p>Download the <code>getting-started.R</code> script from here: <a
href="https://github.com/nsf-dart/RBioTools/blob/91a28f0bc261617280a63ef0dbd4b712cbfb0200/docs/getting-started.html"
class="uri">https://github.com/nsf-dart/RBioTools/blob/91a28f0bc261617280a63ef0dbd4b712cbfb0200/docs/getting-started.html</a></p>
<ul>
<li>When you click on the above link the code will open in browser
window; right click in that window and select, “Save as…” (or “Save page
as …”); make sure that the save as type is set to “Text Document”</li>
<li>Save this R script in the root of your Working Directory</li>
</ul></li>
<li><p>Check the version of R that is being used. These instructions
were prepared using R version 4.1.2 (2021-11-01, “Bird Hippie”); older
or newer versions may behave differently.</p>
<ul>
<li>You can check your version by running <code>sessionInfo()</code> in
your R Console.</li>
</ul></li>
<li><p>Make sure you are connected to the Internet</p></li>
<li><p>Open R Studio</p>
<ul>
<li>Go to File –&gt; Open File –&gt; Navigate to your working folder;
and open “getting-started.R”</li>
</ul></li>
</ol>
</div>
<div id="install-the-required-packages-and-rbiotools"
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Install the required
packages and RBiotools</h1>
<ol style="list-style-type: decimal">
<li>Install some basic dependencies that are required for
<code>RBiotools</code></li>
</ol>
<pre class="r"><code>  #Install some basic dependencies that are required for RBiotools
  install.packages(c(&quot;rlang&quot;,&quot;ape&quot;, &quot;data.table&quot;, &quot;fmsb&quot;, &quot;gplots&quot;, &quot;ggplot2&quot;, &quot;grImport&quot;, &quot;gridExtra&quot;, &quot;pheatmap&quot;, &quot;RCurl&quot;, &quot;rentrez&quot;, &quot;seqinr&quot;))
  
  install.packages(c(&quot;lattice&quot;,&quot;mgcv&quot;,&quot;nlme&quot;,&quot;survival&quot;))</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Install <code>msa</code> and <code>Biostrings</code> package via
<code>BiocManager</code> (and install <code>BiocManager</code> if it is
not already installed)</li>
</ol>
<pre class="r"><code>  #Install msa and Biostrings package via BiocManager (and install BiocManager if it is not already installed)
  if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
      install.packages(&quot;BiocManager&quot;, repos =&quot;http://cran.us.r-project.org&quot;)

  BiocManager::install(&quot;msa&quot;, force = TRUE)
  
  BiocManager::install(&quot;Biostrings&quot;, force = TRUE)</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Install <code>installr</code> - needed to compile
<code>RBiotools</code> from tar.gz</li>
</ol>
<pre class="r"><code>  #Install installr - needed to load RBiotools from tar.gz
  install.packages(&quot;installr&quot;, repos =&quot;http://cran.us.r-project.org&quot;)</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Load the following libraries:</li>
</ol>
<pre class="r"><code>  ## this library call is required so the dendrogram will work
  library(msa) 
  
  
  ## this library call is required so we can install RBiotools from the zip source
  library(installr) </code></pre>
<ol start="5" style="list-style-type: decimal">
<li><p>Set the Working Directory</p>
<ul>
<li><p><em>Setting the working directory:</em> In 1.1.4 of the previous
section, <a href="#install-r-r-studio-and-other-setup">Install R, R
Studio, and other setup</a>, you designated a folder on your computer as
your <strong>Working Directory.</strong></p></li>
<li><p>In this step we assign this as the working directory in R. Be
sure to change the code below, as needed, so that the path is
correct.</p></li>
<li><p><em>Note:</em> For these instructions we have placed the BMIG5101
folder in our C:/Temp directory as demonstration. Please make sure you
select a place that makes sense in your file structure.</p></li>
</ul></li>
</ol>
<pre class="r"><code>setwd(&quot;C://Temp//RBiotools//&quot;)</code></pre>
<pre><code>## [1] &quot;Folder exists already&quot;</code></pre>
<ol start="6" style="list-style-type: decimal">
<li><p>Install the <code>RBiotools</code> package</p>
<p>We will install <code>RBioTools</code> from zip source. Note: you may
need to update the file path depending on where you saved the tar.gz
from <a href="#install-r-r-studio-and-other-setup">Install R, R Studio,
and other setup</a>; step 5.</p></li>
</ol>
<pre class="r"><code>  install.packages(&quot;RBiotools_0.5.3.tar.gz&quot;, repos = NULL, type = &quot;source&quot;)</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Load the <code>RBiotools</code> library</li>
</ol>
<pre class="r"><code>  library(RBiotools) </code></pre>
<hr />
</div>
<div id="identify-the-organisms" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Identify the
organisms</h1>
<p>The first step is to choose a set of organisms that you would like to
explore.</p>
<p>What is an organism?</p>
<ul>
<li>Organisms are specified with GenBank identifiers (aka. accession
IDs) Genbank: <a href="https://www.ncbi.nlm.nih.gov/genbank/"
class="uri">https://www.ncbi.nlm.nih.gov/genbank/</a></li>
<li>Organisms NOT in Genbank can be added with the RBiotools addGenome
function</li>
</ul>
<div id="create-a-list-of-organisms-of-interest" class="section level2"
number="3.1">
<h2><span class="header-section-number">3.1</span> Create a list of
organisms of interest</h2>
<p>Let’s use a sample sets of organisms to get started.</p>
<ol style="list-style-type: decimal">
<li>Create a list of E. coli organisms. This set will be stored using
the name <code>eColi</code>.</li>
</ol>
<pre class="r"><code>eColi &lt;- c(
  &quot;AP012306&quot;,  # Escherichia coli str. K-12 substr. MDS42 DNA         3.976 Mb - smallest chromosome
  &quot;KK583188&quot;,  # Escherichia coli DSM 30083 = JCM 1649 = ATCC 11775   4.907 Mb - type strain scaffold
  &quot;U00096&quot;,    # Escherichia coli str. K-12 substr. MG1655            4.642 Mb - first E. coli genome sequenced
  &quot;CP000802&quot;,  # Escherichia coli HS                                  4.644 Mb - group A representative, commensal
  &quot;CP000800&quot;,  # Escherichia coli E24377A                             4.980 Mb - group B1 representative
  &quot;AP009378&quot;,  # Escherichia coli SE15                                4.717 Mb - group B2 representative, commensal
  &quot;FM180568&quot;,  # Escherichia coli 0127:H6 E2348/69                    4.966 Mb - group B2 representative, enteropathogenic
  &quot;CU928163&quot;,  # Escherichia coli UMN026                              5.202 Mb - group D representative
  &quot;CP008957&quot;,  # Escherichia coli O157:H7 str. EDL933                 5.547 Mp - group E representative
  &quot;CP027027&quot;,  # Shigella dysenteriae strain E670/74                  5.037 Mb - Shigella dysenteria representative
  &quot;CP026802&quot;,  # Shigella sonnei strain ATCC 29930                    4.975 Mb - Shigella sonnei representative
  &quot;CP026877&quot;,  # Shigella boydii strain ATCC 35964                    5.129 Mb - Shigella boydii representative
  &quot;CP026793&quot;,  # Shigella flexneri strain 74-1170                     4.734 Mb - Shigella flexneri representative
  &quot;CP015831&quot;   # Escherichia coli O157 strain 644-PT8                 5.831 Mb - largest chromosome
)</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Create a list of organisms from the Proteobacteria classes. This set
will be stored using the name <code>proteobacteria</code>.</li>
</ol>
<pre class="r"><code>proteobacteria &lt;- c(
  &quot;CP018228&quot;, # Rhizobium leguminosarum strain Vaf-108              Phylum: Proteobacteria (alpha)
  &quot;CP017405&quot;, # Bordetella pertussis strain J448                    Phylum: Proteobacteria (beta)
  &quot;CP008957&quot;, # Escherichia coli O157:H7 str. EDL933                Phylum: Proteobacteria (gamma)
  &quot;HG530068&quot;, # Pseudomonas aeruginosa PA38182                      Phylum: Proteobacteria (gamma)
  &quot;CP002031&quot;, # Geobacter sulfurreducens KN400                      Phylum: Proteobacteria (delta)
  &quot;CP002332&quot;  # Helicobacter pylori Gambia94/24                     Phylum: Proteobacteria (epsilon)
)</code></pre>
</div>
<div
id="verify-list-items-by-downloading-genomic-information-from-genbank"
class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Verify list items by
downloading genomic information from GenBank</h2>
<p>Once you have a sample set of organisms in a list, use
<code>downloadGenBank()</code> to create a data frame with the details
of the organisms.</p>
<pre class="r"><code>downloadGenBank(eColi)</code></pre>
<pre><code>## Initializing RBiotools
## Initializing dependencies ...
## Downloading genome data for organism with accession ID: AP012306 
## Downloading genome data for organism with accession ID: KK583188 
## Downloading genome data for organism with accession ID: U00096 
## Downloading genome data for organism with accession ID: CP000802 
## Downloading genome data for organism with accession ID: CP000800 
## Downloading genome data for organism with accession ID: AP009378 
## Downloading genome data for organism with accession ID: FM180568 
## Downloading genome data for organism with accession ID: CU928163 
## Downloading genome data for organism with accession ID: CP008957 
## Downloading genome data for organism with accession ID: CP027027 
## Downloading genome data for organism with accession ID: CP026802 
## Downloading genome data for organism with accession ID: CP026877 
## Downloading genome data for organism with accession ID: CP026793 
## Downloading genome data for organism with accession ID: CP015831</code></pre>
<p>The results will be stored in two separate data frames: orgData.df
and orgName.df; both can be opened from the <em>Environment</em>
pane.</p>
<p>Click on the data frame in the Environment pane to view the contents
or type <code>View(orgData.df)</code> directly into the Console.</p>
<p>Follow the same steps to view the contents of the
<code>orgName.df</code> data frame</p>
<p>If you have multiple lists, note that the results are not overwritten
if you run this again but are appended to the same data frame.</p>
<pre class="r"><code>downloadGenBank(proteobacteria)</code></pre>
<pre><code>## Downloading genome data for organism with accession ID: CP018228 
## Downloading genome data for organism with accession ID: CP017405 
## Genome data for accession ID: CP008957 has already downloaded
## Downloading genome data for organism with accession ID: HG530068 
## Downloading genome data for organism with accession ID: CP002031 
## Downloading genome data for organism with accession ID: CP002332</code></pre>
</div>
</div>
<div id="explore-functions-that-make-plots-and-figures"
class="section level1" number="4">
<h1><span class="header-section-number">4</span> Explore functions that
make plots and figures</h1>
<div id="dendrograms" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Dendrograms</h2>
<p>Plotting a dendrogram using 16S rRNA genes is pretty easy; however,
may take several minutes to complete while data is accessed from
GenBank. This will depend on how many items are in your list. Note that
the proteobacteria list is shorter than the eColi list, so it completes
more quickly.</p>
<pre class="r"><code>  #This first plot is for the eColi and Shigella organisms
  dendrogram16S(eColi)</code></pre>
<pre><code>## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: AP012306
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: KK583188
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: U00096
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP000802
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP000800
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: AP009378
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: FM180568
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CU928163
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP008957
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP027027
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP026802
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP026877
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP026793
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP015831
## Creating dendrogram for 14 16S rRNA sequences
## use default substitution matrix</code></pre>
<p><img src="getting-started_files/figure-html/dendrogram.eC-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  #The second plot is for the proteobacteria organisms
  dendrogram16S(proteobacteria)</code></pre>
<pre><code>## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP018228
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP017405
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: HG530068
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP002031
## Calling type &#39;SSU&#39; rRNA genes for genome with accession ID: CP002332
## Creating dendrogram for 6 16S rRNA sequences
## use default substitution matrix</code></pre>
<p><img src="getting-started_files/figure-html/dendrogram.pb-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="genome-atlas" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Genome Atlas</h2>
<p>If you want to create a genome atlas for an organism already in a
list, such as the ones we have already created (<code>eColi</code> and
<code>proteobacteria)</code>, then you can index into the list for that
value.</p>
<pre class="r"><code>  createAtlas(eColi[1]) # here we index into the list to select the first item, AP012306.</code></pre>
<pre><code>## Writing SVG code for genome with accession ID: AP012306 
## Calling protein genes for accession ID: AP012306 
## Calling type TSU rRNA genes for genome with accession ID: AP012306
## Calling type LSU rRNA genes for genome with accession ID: AP012306</code></pre>
<p><img src="R_repository/GenomeAtlas_AP012306.png" /></p>
<p>Note: This graphic generated at this step will be a stored as an SVG
file in your Working Directory. SVG files can be accessed by:</p>
<ul>
<li>Use your File Explorer (or equivalent) to navigate to your Working
Directory</li>
<li>Right-click on the file</li>
<li>Hover over “Open with”</li>
<li>Select a browser (Firefox, Google Chrome, etc…) to open the
file.</li>
</ul>
<div class="figure">
<img src="r_repository/genome-atlas-reference-image.png" width="671"
alt="" />
<p class="caption">Opening an SVG using a web browser.</p>
</div>
<p>If you want to create a genome atlas for an item <strong>not</strong>
already stored in a list, you can set that “on-the-fly” by locating the
accession ID of interest from GenBank and entering as we have below.</p>
<pre class="r"><code>  createAtlas(&quot;AP012306&quot;) # here we hard code the accession ID from GenBank to create a genome atlas for an item not already in a list.</code></pre>
</div>
<div id="amino-acid-usage-plot" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Amino Acid Usage
Plot</h2>
<p>The <code>plotUsage</code> function has many options. See the
<code>plotUsage</code> documentation for more details and
parameters.</p>
<pre class="r"><code>  plotUsage(eColi[1])</code></pre>
<pre><code>## Plotting AA usage across 3621 proteins</code></pre>
<p><img src="getting-started_files/figure-html/plot.usage-1.png" width="672" /></p>
</div>
<div id="codon-heat-map" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Codon Heat Map</h2>
<p>Heat Maps work best for groups of organisms that are NOT closely
related (like our Proteobacteria group)</p>
<pre class="r"><code>  plotHeatMapCodon(proteobacteria)</code></pre>
<pre><code>## Calling protein genes for accession ID: CP018228 
## Calling protein genes for accession ID: CP017405 
## Calling protein genes for accession ID: CP008957 
## Calling protein genes for accession ID: HG530068 
## Calling protein genes for accession ID: CP002031 
## Calling protein genes for accession ID: CP002332</code></pre>
<p><img src="R_repository/HeatMapCodon.png" /></p>
<p>Note: This graphic generated at this step will be a stored as a PNG
file in your Working Directory. This file can be accessed by:</p>
<ul>
<li>Go to the “Files” tab in RStudio (this tab will be in one of your
IDE quadrants)</li>
<li>Navigate to your Working Directory * hint: you may need to refresh
your Working Directory for it to appear</li>
<li>Click on the “HeatMapCodon.png” file</li>
</ul>
<div class="figure">
<img src="R_repository/heat-codon-reference-image.png" width="220"
alt="" />
<p class="caption">Opening the heat codon result from within
RStudio.</p>
</div>
</div>
<div id="blast-matrix" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Blast Matrix</h2>
<p>Blast Matrices work best for groups of closely related organisms
(like our eColi group)</p>
<ol style="list-style-type: decimal">
<li><p>Build a “homology matrix”, a table where:</p>
<ul>
<li><p>each table row is a protein group</p></li>
<li><p>each table column is an organism</p></li>
<li><p>each table entry is the number of an organism’s proteins in a
group</p></li>
</ul></li>
</ol>
<pre class="r"><code>  proteinGrouping &lt;- runLinclust(eColi)</code></pre>
<pre><code>## Calling protein genes for accession ID: KK583188 
## Calling protein genes for accession ID: U00096 
## Calling protein genes for accession ID: CP000802 
## Calling protein genes for accession ID: CP000800 
## Calling protein genes for accession ID: AP009378 
## Calling protein genes for accession ID: FM180568 
## Calling protein genes for accession ID: CU928163 
## Calling protein genes for accession ID: CP027027 
## Calling protein genes for accession ID: CP026802 
## Calling protein genes for accession ID: CP026877 
## Calling protein genes for accession ID: CP026793 
## Calling protein genes for accession ID: CP015831 
## Number of ungapped alignments: 112023
## Number of Smith-Waterman alignments: 2514</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Use the homology matrix to plot the Blast Matrix</li>
</ol>
<pre class="r"><code>  plotBlastMatrix(proteinGrouping)</code></pre>
<p><img src="R_repository/BlastMatrix.png" /></p>
<p>Note: This graphic generated at this step will be a stored as an SVG
file in your Working Directory. SVG files can be accessed by:</p>
<ul>
<li>Use your File Explorer (or equivalent) to navigate to your Working
Directory</li>
<li>Righ-click on the file</li>
<li>Hover over “Open with”</li>
<li>Select a browser (Firefox, Google Chrome, etc…) to open the
file.</li>
</ul>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
